packages/generator/src/parser/edmx-to-vdm-v4.ts

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { VdmNavigationProperty, VdmComplexType, VdmEntity } from '../vdm-types';
+import { ServiceNameFormatter } from '../service-name-formatter';
+import { EdmxEntityType, EdmxEntitySet } from './parser-types-v4';
+import { isCollection } from './parser-util';
+import {
+  JoinedEntityMetadata,
+  ParsedServiceMetadata
+} from './parser-types-common';
+import {
+  joinEntityMetadata,
+  createEntityClassNames,
+  transformEntity,
+  navigationPropertyBase
+} from './edmx-to-vdm-common';
+
+function navigationProperties(
+  entityMetadata: JoinedEntityMetadata,
+  classNames: { [originalName: string]: string },
+  formatter: ServiceNameFormatter
+): VdmNavigationProperty[] {
+  const entityType = entityMetadata.entityType as EdmxEntityType;
+  const entitySet = entityMetadata.entitySet as EdmxEntitySet;
+
+  return entitySet.NavigationPropertyBinding.map(navBinding => {
+    const navProp = entityType.NavigationProperty.find(
+      n => n.Name === navBinding.Path
+    );
+
+    if (!navProp) {
+      throw new Error(
+        `Could not find navigation property ${navBinding.Path} in entity type ${entityType.Name}.`
+      );
+    }
+
+    const isCollectionType = isCollection(navProp.Type);
+
+    return {
+      ...navigationPropertyBase(
+        navProp.Name,
+        entityMetadata.entitySet.Name,
+        formatter
+      ),
+      from: entityMetadata.entityType.Name,
+      to: navBinding.Target,
+      toEntityClassName: classNames[navBinding.Target],
+      multiplicity: isCollectionType ? '1 - *' : '1 - 1',
```

Comment:
What about an enum? Is the string used somewhere and if so where? Also the `isMultiLink` not redundant information?

Reply:
or multiplicity is redundant if you have the boolean.

Reply:
Yeah, totally, that is why I deprecated it. It isn't even used anywhere. I still left it here, so that it is consistent and less of a breaking change.